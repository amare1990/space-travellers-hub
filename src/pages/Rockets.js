import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets, reserveRocket, cancelRocket } from '../redux/rockets/rockets';
import '../styles/Rocket.css';

function Rockets() {
  const rocketsArray = useSelector((state) => state.rockets.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    if (rocketsArray.length === 0) {
      dispatch(fetchRockets());
    }
  });

  return (

    <div className="rockets-wrapper">
      { rocketsArray.map((rocket) => (

        <li key={rocket.rocket_id} className="rocket-card">
          <img
            src={rocket.flickr_images[0]}
            alt={rocket.rocket_name}
            className="rocket-img {rocket.rocket_name}"
          />
          <div className="rocket-body">
            <p className="rocket-title">
              {rocket.rocket_name}
            </p>
            <p className="rocket-description">
              {rocket.reserved ? (
                <span className="reserved-text">
                  Reserved
                </span>
              )
                : ''}
              {rocket.description}
            </p>

            {rocket.reserved
              ? (
                <button
                  type="button"
                  className="btn reserve-rocket-btn"
                  onClick={() => dispatch(cancelRocket(rocket.rocket_id))}
                >
                  Cancel Reservation
                </button>
              )
              : (
                <button
                  type="button"
                  className="btn reserve-rocket-btn"
                  onClick={() => dispatch(reserveRocket(rocket.rocket_id))}
                >
                  Reserve Rocket
                </button>
              )}

          </div>

        </li>
      ))}
    </div>
  );
}

export default Rockets;
