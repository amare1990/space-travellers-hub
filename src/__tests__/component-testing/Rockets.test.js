import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../../pages/Rockets';
import store from '../../redux/configureStore';

test('Testing if Rockets component renders correctly', () => {
  const rocketsSnapshot = renderer.create(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  ).toJSON();
  expect(rocketsSnapshot).toMatchSnapshot();
});
