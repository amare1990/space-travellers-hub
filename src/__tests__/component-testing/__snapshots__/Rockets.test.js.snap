// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if Rockets component renders correctly 1`] = `
<div
  className="rockets-wrapper"
/>
`;
